src/bp/sdk/botpress.d.ts

```diff
@@ -457,6 +457,71 @@ declare module 'botpress/sdk' {
   }
 
   export namespace NLU {
+    export class Engine {
+      static initialize: (config: Config, logger: NLU.Logger) => Promise<void>
+      static getHealth: () => Health
+      static getLanguages: () => string[]
+      constructor(defaultLanguage: string, botId: string, logger: Logger)
+      computeModelHash(intents: NLU.IntentDefinition[], entities: NLU.EntityDefinition[], lang: string): string
+      loadModel: (m: Model) => Promise<void>
+      train: (
+        intentDefs: NLU.IntentDefinition[],
+        entityDefs: NLU.EntityDefinition[],
+        languageCode: string,
+        reportTrainingProgress?: ProgressReporter,
+        trainingSession?: TrainingSession,
+        options?: TrainingOptions
+      ) => Promise<Model | undefined>
+      predict: (t: string, ctx: string[]) => Promise<IO.EventUnderstanding>
+    }
+
+    export interface Config {
+      ducklingURL: string
+      ducklingEnabled: boolean
+      languageSources: LanguageSource[]
+    }
+
+    export interface LanguageSource {
+      endpoint: string
+      authToken?: string
+    }
+
+    export interface Logger {
```

Comment:
It keeps bugging me to have a different, simplified logger interface. I understand why but it bugs me, should we rename it ?

Reply:
NLULogger? EngineCallback? NLUCallback?

I can also remove the interface and change declare Engine like this:

```
export class Engine {
      static initialize: (config: Config, logger: NLU.Logger) => Promise<void>
      static getHealth: () => Health
      static getLanguages: () => string[]
      constructor(defaultLanguage: string, botId: string, logger: { info: (msg: string) => void
                                                                    warning: (msg: string, err?: Error) => void
                                                                    error: (msg: string, err?: Error) => void })
```